tensorboard/plugins/projector/vz_projector/vz-projector-inspector-panel.ts

```diff
@@ -127,6 +169,14 @@ namespace vz_projector {
       this.enableResetFilterButton(false);
     }
 
+    _updateNeighborsList() {
```

Comment:
Can this be inlined?

Reply:
Wait I'm not totally sure what you mean inlined.

This?
```javascript
      showNeighborImages: {
        type: Boolean,
        value: true,
        observer: () => {
          this.updateNeighborsList();
        },
      },
```
this?
```javascript
    _updateNeighborsList() { this.updateNeighborsList(); }
```

Doing this doesn't work because it's not bound to `this` anymore:
```javascript
      showNeighborImages: {
        type: Boolean,
        value: true,
        observer: 'updateNeighborsList',
      },
```

Reply:
Ah, I see what’s happening. By “inlined” I meant “remove the function
`_updateNeighborsList` and replace all calls with `updateNeighborsList`,
which should ‘obviously’ be equivalent”. But the reason that this
doesn’t work is that Polymer invokes observers with the value of the
property, which for `showNeighborImages` is a boolean—so, when the
checkbox is clicked, Polymer calls `_updateNeighborsList(true)`, which
then calls `updateNeighborsList()`. After inlining, Polymer will instead
pass that argument to `updateNeighborsList` directly, but then it’s
interpreted as the value of `neighbors?: knn.NearestEntry[]` instead of
being ignored, which leads to errors.

The implementation is fine, but the above is confusing enough that the
function should have a comment and probably a rename—something like
`_refreshNeighborsList`, with a comment indicating that it specifically
drops all arguments before calling `updateNeighborsList`.